src/www/app/outline_server_repository/server.ts

```diff
@@ -96,10 +120,33 @@ export class OutlineServer implements Server {
     return this.accessKey.includes('outline=1');
   }
 
+  private async startDynamicConfigRefresh() {
+    this.stopDynamicConfigRefresh();
+    const callback = async () => {
+      const sessionConfigs =
+        await fetchShadowsocksSessionConfig(this.sessionConfigLocation);
+      this.configQueue.updateConfigs(sessionConfigs)
+    }
+    await callback();
+    this.dynamicConfigInterval = setInterval(
+      callback, 5 /*min*/ * 60 /*sec/min*/ * 1000 /*ms/sec*/);
+  }
+
+  private stopDynamicConfigRefresh() {
+    if (!this.dynamicConfigInterval) return;
+    clearInterval(this.dynamicConfigInterval);
+    this.dynamicConfigInterval = null;
+  }
+
   async connect() {
+    // connect() is triggered by a manual user connect action
+    // so we must reset the configQueue.
+    this.configQueue.reset();
     try {
       if (this.type === ServerType.DYNAMIC_CONNECTION) {
-        this.sessionConfig = await fetchShadowsocksSessionConfig(this.sessionConfigLocation);
+        this.startDynamicConfigRefresh();
+        this.sessionConfig = this.configQueue.getConfig();
```

Comment:
I don't understand. What config is this returning?
Is your code doing a round-robin queue, so each click on connect gets the next config?